Starting Point (#2)

zahifarah opened the pull request:
Okay! So, here we go, a _"significant"_ update. 

1- Got a seed file in which there's an object we can add/remove key-value pairs to/from.
2- fetchPrices() iterates over every property of that seed file list and fetches the relevant data for that pair.
3- Automatically display some of the returned information in "home.ejs"

A lot remains to be done of course, but this is at least a crude, kind-of working starting point.

Cheers 🍻

zahifarah committed:
Figuring out stuff :)

zahifarah committed:
Figuring out how to manipulate JSON data into workable a workable array.

zahifarah committed:
Work in progress.

zahifarah committed:
fetchPrices function now works as intended. Iterates over a predefined list in seed.js file.

zahifarah committed:
Rename seedHelper.js to seed.js

zahifarah committed:
Automate card creation.

zahifarah committed:
Add gradient to page title.

raminos commented on index.js:
```diff
@@ -1,43 +1,35 @@
-  "https://api.cryptonator.com/api/ticker/eth-usd",
-  "https://api.cryptonator.com/api/ticker/ada-usd",
-]
+let cryptoData = {}
```
Did you know that you can define objects and Arrays almost always directly as `const` even when you change the values inside? If you didn't find out why :D

```suggestion
const cryptoData = {}
```

zahifarah replied:
Yes! Something I learned in the advanced JS course (that I'm still taking) is that Objects and Arrays are passed by reference, as such changing _what_ they're storing doesn't affect their place in memory. Problems/bugs, and some of the ensuing confusion it can cause, can arise when someone attempts to re-assign or re-declare something they shouldn't.




I think.

raminos commented on seed/seed.js:
```diff
@@ -0,0 +1,9 @@
+const cryptoPairs = {
```
We often use names like `constants` for these kinds of data. You don't need a folder. You could just start with a file called `constants.js`. When the project gets bigger the folder can be called `constants` and the file based on the information `currencies.js` or `cryptoPairs.js`

raminos reviewed (approved):
Great work!

I see that your commit messages are... quite random 😄 . If you want to learn more on how to write good commit messages look at this: https://chris.beams.io/posts/git-commit/ (it's quite long you don't need to read all of it).

Also, could you add to the project how to run it and what package manager to use and which secrets to add (if any). I didn't succeed at running the project. I get an error: `Cannot GET /`

zahifarah committed:
Use rems on page-title for a more responsive layout

zahifarah committed:
Rename seed.js to cryptoPairs.js for clarity

zahifarah committed:
Change cryptoData to a constant

zahifarah committed:
Change fetchPrices function to become Promise.all based

Instead of using multiple, individual fetch calls, wrap them in a Promise.all method for a more optimized performance.

raminos commented:
Great commit messages! 🥇
